topi/include/topi/transform.h

```diff
@@ -365,6 +365,84 @@ inline Array<Tensor> split(const Tensor& x,
   return result;
 }
 
+/*!
+* \brief strided_slice of a tensor
+*
+* \param x The input tensor
+* \param begin The indices to begin with in the slicing
+* \param end Indicies indicating end of the slice
+* \param strides Specifies the stride values, it can be negative
+* in that case, the input tensor will be reversed in that particular axis
+* \param name The name of the operation
+* \param tag The tag to mark the operation
+*
+* \return A Tensor whose op member is the split operation
+*/
+inline Tensor strided_slice(const Tensor& x,
+                            const Array<Expr>& begin,
+                            const Array<Expr>& end,
+                            const Array<Expr>& strides,
+                            std::string name = "tensor",
+                            std::string tag = kInjective) {
+  size_t src_tensor_dim = static_cast<size_t>(x->shape.size());
+  std::vector<int> begin_vec = GetConstIntValues(begin, "begin");
+  std::vector<int> end_vec = GetConstIntValues(end, "end");
```

Comment:
The way integer works is that we always store constant index in int64_t when possible if it involves index. When we use these integers to construct the index expressions. Do

```c++
int64_t begin_index;
Expr begin_expr = make_const(begin[i].type(), begin_index);
// Use begin_expr in index calculation
```

In this way, when user constructs index expression in type int64, the index expression type will remain int64. But when user uses int32, the index expression will be int32(although the constant value in constructing these expressions are stored in int32).

Please also add a test case that uses int64 as index and stride type

Reply:
int64 can be constructed via tvm.const(value, "int64")

Reply:
see more comments on http://docs.tvm.ai/contribute/code_guide.html#handle-integer-constant-expression

Reply:
@tqchen : I have modified code according to your comments.
But when i add a testcase to form "int64" type as begin, end & stride i get below error
File "/home/anshuman/work/nnvm-commit/tvm-latest/tvm/python/tvm/tag.py", line 33, in tagged_fdecl
    return func(*args, **kwargs)
  File "/home/anshuman/work/nnvm-commit/tvm-latest/tvm/topi/python/topi/transform.py", line 156, in strided_slice
    return cpp.strided_slice(a, begin, end, strides)
  File "/home/anshuman/work/nnvm-commit/tvm-latest/tvm/python/tvm/_ffi/function.py", line 280, in my_api_func
    return flocal(*args)
  File "/home/anshuman/work/nnvm-commit/tvm-latest/tvm/python/tvm/_ffi/_ctypes/function.py", line 183, in __call__
    ctypes.byref(ret_val), ctypes.byref(ret_tcode)))
  File "/home/anshuman/work/nnvm-commit/tvm-latest/tvm/python/tvm/_ffi/base.py", line 66, in check_call
    raise TVMError(py_str(_LIB.TVMGetLastError()))
tvm._ffi.base.TVMError: [19:10:41] src/ir/IR.cpp:465: Check failed: args[i].type() == Int(32) Args to call to halide function must be type Int(32)

Would you please help me, how to resolve this error.
It seems "int64" not supported in const building currently.

Reply:
OK, we can leave the that test-cases out for now, but please fix other comments